File: airbyte-integrations/bases/base-java-s3/src/test/java/io/airbyte/integrations/destination/s3/writer/BaseS3WriterTest.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.s3.writer;

import static org.junit.jupiter.api.Assertions.assertEquals;

import io.airbyte.integrations.destination.s3.S3Format;
import io.airbyte.integrations.destination.s3.template.S3FilenameTemplateParameterObject;
import java.io.IOException;
import java.sql.Timestamp;
import org.junit.jupiter.api.Test;

class BaseS3WriterTest {

  @Test
  public void testGetOutputFilename() throws IOException {
    final Timestamp timestamp = new Timestamp(1471461319000L);
    assertEquals(
        "2016_08_17_1471461319000_0.csv",
        BaseS3Writer.determineOutputFilename(S3FilenameTemplateParameterObject
            .builder()
            .s3Format(S3Format.CSV)
            .timestamp(timestamp)
            .build()));
  }

}

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/test/kotlin/io/airbyte/cdk/integrations/destination/s3/writer/BaseS3WriterTest.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.s3.writer

import io.airbyte.cdk.integrations.destination.s3.FileUploadFormat
import io.airbyte.cdk.integrations.destination.s3.template.S3FilenameTemplateParameterObject.Companion.builder
import io.airbyte.cdk.integrations.destination.s3.writer.BaseS3Writer.Companion.determineOutputFilename
import java.io.IOException
import java.sql.Timestamp
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class BaseS3WriterTest {
    @Test
    @Throws(IOException::class)
    fun testGetOutputFilename() {
        val timestamp = Timestamp(1471461319000L)
        Assertions.assertEquals(
            "2016_08_17_1471461319000_0.csv",
            determineOutputFilename(
                builder().s3Format(FileUploadFormat.CSV).timestamp(timestamp).build()
            )
        )
    }
}
